workbench-core/base/src/utilities/queryParamsFiltersUtils.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { marshall } from '@aws-sdk/util-dynamodb';
+import Boom from '@hapi/boom';
+import { QueryParams } from '../aws/helpers/dynamoDB/dynamoDBService';
+import { FilterRequest } from '../interfaces/FilterRequest';
+import { SortRequest } from '../interfaces/SortRequest';
+
+/************************************************************
+ * Validates only one sorting or filter in list requests
+ * @param filter - filter request object
+ * @param sort - sorting request object
+ ************************************************************/
+export function validateSingleSortAndFilter(filter?: FilterRequest, sort?: SortRequest): void {
+  const filterAttributesLength = filter ? Object.keys(filter).length : 0;
+  const sortAttributesLength = sort ? Object.keys(sort).length : 0;
+  // Check that at most one filter is defined because we not support more than one filter
+  if (filterAttributesLength > 1) {
+    throw Boom.badRequest('Cannot apply more than one filter.');
+  }
+  // Check that at most one sort attribute is defined because we not support sorting by more than one attribute
+  if (sortAttributesLength > 1) {
+    throw Boom.badRequest('Cannot sort by more than one attribute.');
+  }
+  const hasSameProperty =
+    filterAttributesLength === 1 &&
+    sortAttributesLength === 1 &&
+    Object.keys(filter ?? {})[0] === Object.keys(sort ?? {})[0];
+  if (filter && sort && !hasSameProperty) {
+    throw Boom.badRequest('Cannot apply a filter and sort at the same time');
+  }
+}
+
+/************************************************************
+ * Parses FilterRequest into QueryParam object consumed by dynamoDB
+ *
+ * @param filter - filter request object
+ * @param gsiPropertyNames - field names on pascal case format to retrieve gsi name
+ *
+ * @returns QueryParams object
+ ************************************************************/
+export function getFilterQueryParams(filter: FilterRequest | undefined, gsiNames: string[]): QueryParams {
+  let queryParams: QueryParams = {};
+  //validate filter request doesnt have more than one filter or no filters
+  if (!filter || Object.keys(filter).length === 0) {
+    return queryParams;
+  }
+  if (Object.keys(filter).length > 1) {
+    throw Boom.badRequest('Cannot filter by more than one attribute.');
+  }
+  //only one entry will exist due to validation above
+  Object.entries(filter).forEach(([key, value]) => {
+    //transform { operator: value } into { operator { type: value } }
+    const filterQueryParam = (
+      value?.between ? { between: marshall(value.between) } : marshall(value)
```

Comment:
this line make me think that the `QueryParameterFilter` can either have `between` or the rest of the fields (`lt`, `lte`, `eq`, etc) defined. Is that correct?

Reply:
well, we can only have one type of filter, dynamo will complain if we have more than one, but I had a problem applying marshall to between, I will add an extra type , so I had to send only what is inside between to make it work , other operators can just be marshall and will work

Reply:
I dont remember seeing documentation saying that only one field can be defined at a time. Probably worth adding to the interface

Reply:
added warning to interface explaining we can only use one operator